src/prefect/blocks/abstract.py

```diff
@@ -302,3 +302,61 @@ async def upload_from_folder(
         Returns:
             The path that the folder was uploaded to.
         """
+
+
+class SecretBlock(Block, ABC):
+    """
+    Block that represents a resource that can store and retrieve secrets.
+    """
+
+    @property
+    def logger(self) -> Logger:
+        """
+        Returns a logger based on whether the SecretBlock
+        is called from within a flow or task run context.
+        If a run context is present, the logger property returns a run logger.
+        Else, it returns a default logger labeled with the class's name.
+        """
+        try:
+            return get_run_logger()
+        except MissingContextError:
+            return get_logger(self.__class__.__name__)
+
+    @abstractmethod
+    async def read_secret(self):
+        """
+        Reads the secret from the secret storage service.
+        """
+
+    @abstractmethod
+    async def write_secret(self, secret_data) -> str:
+        """
+        Writes the secret to the secret storage service.
+
+        Args:
+            secret_data: The secret data to write.
+
+        Returns:
+            The path that the secret was written to.
+        """
+
+    @abstractmethod
+    async def update_secret(self, secret_data) -> str:
+        """
+        Updates the secret to the secret storage service.
+
+        Args:
+            secret_data: The secret data to update.
+
+        Returns:
+            The path that the secret was updated to.
+        """
+
+    @abstractmethod
+    async def delete_secret(self) -> str:
```

Comment:
I'd vote for removing this one also. I don't know how often users are deleting secrets via their flows and I don't want to require implementing a method that may go completely unused. It doesn't preclude implementation from including delete functionality, and we can add it later if there's an emergent pattern among implementation of this ABC.